### PR TITLE
Branch handle NumberFormatException and allow numerical values in description (Edit Command)

### LINKED ISSUE #153: [PE-D] EDIT command feature flaw
My input : `EDIT 1 i/3 boxes of milk`

The command hasn't been executed correctly

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: DDzuikeai/ped#6

### LINKED ISSUE #155: [PE-D] `EDIT` command feature flaw
### When I type to edit an item in the list, I give correct `name description` but incorrect `price description`, the app doesn't work as I expected.
Initially, this is what is in the shopping list:

My input : `EDIT 1 i/water p/1.0.0`
Output : 

Seems to be correct, but when I enter `DISPLAY`, the output is:

So even the command is incorrect, it's still executed. The item has been changed, but the app didn't give user clear feedback.

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: DDzuikeai/ped#4

### LINKED ISSUE #156: [PE-D] `EDIT` command hasn't handled possilble execptions carefully
### Program crashed when I type incorrect index for an item
My input : `EDIT 1.0 i/beef`
Output :  

When I rerun the program, all items in the list are gone.

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: DDzuikeai/ped#3

### PR DESCRIPTION
**Pull request by trishaangelica**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -168,10 +168,10 @@ have been omitted from the diagram. Those details are shown in a separate sequen
...
  
  
-- Alternative 2: Require all values of an <code>item</code> object to be updated
-  - Pros: Will have less code to deal with having no additional parsing of input string needed.
+- Alternative 2: Require all values of an <code>item</code> object to be updated.
+  - Pros: Will have less code to deal with having no additional parsing of the input string.
   
-  - Cons: Less user flexibility; user must input all parameters even if he/she does not want to update certain
+  - Cons: Less user flexibility; user must input all parameters even if he/she does not wish to update certain
   variables.
```
Comment (chaosInTheWild) [inline docs/DeveloperGuide.md:181]: Maybe can add a statement to explain why we choose alternative 1 over alternative 2 as mentioned in the DG review.

File: src/main/java/seedu/duke/utils/Parser.java [hunk 2]
```diff
@@ -469,31 +470,45 @@ private boolean singleDelimiterChecker(String stringToCheck) {
...
 
             if (matcher.group("description") != null) {
-                itemDescription = matcher.group("description");
+                itemDescription = matcher.group("description").trim();
             }
 
             if (matcher.group("price") != null) {
-                itemPrice = matcher.group("price");
+                if (matcher.group(0).contains("-")) {
+                    throw new NumberFormatException();
```
Comment (chaosInTheWild) [inline src/main/java/seedu/duke/utils/Parser.java]: Perhaps you can just have an if statement to check whether the result of the parseDouble(itemPrice) is lesser than 0, if yes then return new IncorrectCommand?

**Review by chaosInTheWild — COMMENTED** [review]
Generally it looks good to merge, just that maybe can change some parts of the code or user guide to make it flow more naturally :+1:

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (trishaangelica) [reply docs/DeveloperGuide.md:181]: okie thank you will do it!

File: src/main/java/seedu/duke/utils/Parser.java (hunk 2 above)
Comment (trishaangelica) [reply src/main/java/seedu/duke/utils/Parser.java]: okiee!! :D

**Pull request merged**